java-frontend/src/main/java/org/sonar/java/JavaFrontend.java

```diff
@@ -122,9 +124,9 @@ boolean analysisCancelled() {
     return sonarComponents != null && sonarComponents.analysisCancelled();
   }
 
+
   public void scan(Iterable<InputFile> sourceFiles, Iterable<InputFile> testFiles, Iterable<? extends InputFile> generatedFiles) {
     if (isBatchModeEnabled()) {
```

Comment:
We should never use batch mode when `sonarComponents.isSonarLintContext()` because it uses `inputFile.absolutePath()` and not `inputFile.contents()`.